server/channels/api4/channel.go

```diff
@@ -385,6 +385,25 @@ func patchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Validate channel type change if attempted
```

Comment:
I believe after this change the `updateChannelPrivacy` API won't be used any more? If thats right, can you file a ticket for its removal?